### DISCUSSION TITLE
next 15 IIS application server next/link appends .txt to url

### DISCUSSION THREAD

**Discussion by gonzjed1** [vercel/next.js#72045] [Help]
### Summary

Link href ="/testpage" key={'AAP1_test'}  style={{ textDecoration: 'none' }}>Test3 using next/link to href="/testpage"
when navigating to test 3 i get the .txt.html being appended i am using app routing with outpu: 'export'

Application server 
-IIS Server 
-Using SSG 
-React 18 

Maybe web.config issue ? 

### Additional information

```js
<Link href ="/testpage" key={'AAP1_test'}  style={{ textDecoration: 'none' }}>Test3 using next/link to href="/testpage"</Link>
```

**Comment by aniruddhaadak80** [comment]
Hey @gonzjed1 ,

In Next.js, using IIS with static site generation (SSG) may cause unexpected file extensions (like `.txt.html`) to appear if IIS is misconfigured. This often stems from the `.html` extension handling in the `web.config` file, which controls URL rewriting on IIS.

### Solution:
1. **Check `web.config`:** Add or adjust rewrite rules to ensure URLs don’t append `.txt` or other unwanted extensions.
   ```xml
   <rewrite>
       <rules>
           <rule name="RemoveExtensions" stopProcessing="true">
               <match url="(.*)" />
               <conditions>
                   <add input="{REQUEST_FILENAME}" matchType="IsFile" negate="true" />
               </conditions>
               <action type="Rewrite" url="{R:1}" />
           </rule>
       </rules>
   </rewrite>
   ```

2. **Link Component:** Double-check the `<Link>` component. Use absolute paths or ensure `href` values don’t include extensions:
   ```jsx
   <Link href="/testpage" style={{ textDecoration: 'none' }}>Test3</Link>
   ```

3. **SSG Configuration:** Ensure `next.config.js` is correctly set for SSG.

This setup should prevent unwanted extensions and ensure proper routing with IIS and Next.js.

Thanks .
